# Jupyter Notebook for the exploration and the cleaning of the dataset

##### Import libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

##### Import dataset

In [29]:
raw_data = pd.read_csv('../data/investments_VC.csv', encoding='ISO-8859-1')

##### First check of the dataset

In [30]:
raw_data.shape

(54294, 39)

In [31]:
raw_data.head(2)

permalink                name  \
0            /organization/waywire            #waywire   
1  /organization/tv-communications  &TV Communications   

             homepage_url                               category_list  \
0  http://www.waywire.com  |Entertainment|Politics|Social Media|News|   
1   http://enjoyandtv.com                                     |Games|   

   market   funding_total_usd      status country_code state_code  \
0    News           17,50,000    acquired          USA         NY   
1   Games           40,00,000   operating          USA         CA   

          region  ... secondary_market  product_crowdfunding round_A round_B  \
0  New York City  ...              0.0                   0.0     0.0     0.0   
1    Los Angeles  ...              0.0                   0.0     0.0     0.0   

  round_C  round_D round_E round_F  round_G  round_H  
0     0.0      0.0     0.0     0.0      0.0      0.0  
1     0.0      0.0     0.0     0.0      0.0      0.0  

[2 rows x 39 columns]

In [32]:
raw_data.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', ' market ',
       ' funding_total_usd ', 'status', 'country_code', 'state_code', 'region',
       'city', 'funding_rounds', 'founded_at', 'founded_month',
       'founded_quarter', 'founded_year', 'first_funding_at',
       'last_funding_at', 'seed', 'venture', 'equity_crowdfunding',
       'undisclosed', 'convertible_note', 'debt_financing', 'angel', 'grant',
       'private_equity', 'post_ipo_equity', 'post_ipo_debt',
       'secondary_market', 'product_crowdfunding', 'round_A', 'round_B',
       'round_C', 'round_D', 'round_E', 'round_F', 'round_G', 'round_H'],
      dtype='object')

In [33]:
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54294 entries, 0 to 54293
Data columns (total 39 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   permalink             49438 non-null  object 
 1   name                  49437 non-null  object 
 2   homepage_url          45989 non-null  object 
 3   category_list         45477 non-null  object 
 4    market               45470 non-null  object 
 5    funding_total_usd    49438 non-null  object 
 6   status                48124 non-null  object 
 7   country_code          44165 non-null  object 
 8   state_code            30161 non-null  object 
 9   region                44165 non-null  object 
 10  city                  43322 non-null  object 
 11  funding_rounds        49438 non-null  float64
 12  founded_at            38554 non-null  object 
 13  founded_month         38482 non-null  object 
 14  founded_quarter       38482 non-null  object 
 15  founded_year       

##### Observations:
- Shape: 54294 rows, 39 columns
- Columns: 'permalink', 'name', 'homepage_url', 'category_list', ' market ', ' funding_total_usd ', 'status', 'country_code', 'state_code', 'region', 'city',   'funding_rounds', 'founded_at', 'founded_month', 'founded_quarter', 'founded_year', 'first_funding_at', 'last_funding_at', 'seed', 'venture', 'equity_crowdfunding', 'undisclosed', 'convertible_note', 'debt_financing', 'angel', 'grant', 'private_equity', 'post_ipo_equity', 'post_ipo_debt', 'secondary_market', 'product_crowdfunding', 'round_A', 'round_B', 'round_C', 'round_D', 'round_E', 'round_F', 'round_G', 'round_H'

##### Some value counts

In [34]:
raw_data['status'].value_counts()

status
operating    41829
acquired      3692
closed        2603
Name: count, dtype: int64

In [35]:
raw_data['country_code'].value_counts()

country_code
USA    28793
GBR     2642
CAN     1405
CHN     1239
DEU      968
       ...  
MUS        1
SOM        1
CIV        1
SYC        1
MAF        1
Name: count, Length: 115, dtype: int64

In [43]:
# country code = ITA
raw_data[raw_data['country_code'] == 'ITA'].shape[0]

308

In [47]:
raw_data.isnull().sum()

permalink                4856
name                     4857
homepage_url             8305
category_list            8817
market                   8824
funding_total_usd        4856
status                   6170
country_code            10129
state_code              24133
region                  10129
city                    10972
funding_rounds           4856
founded_at              15740
founded_month           15812
founded_quarter         15812
founded_year            15812
first_funding_at         4856
last_funding_at          4856
seed                     4856
venture                  4856
equity_crowdfunding      4856
undisclosed              4856
convertible_note         4856
debt_financing           4856
angel                    4856
grant                    4856
private_equity           4856
post_ipo_equity          4856
post_ipo_debt            4856
secondary_market         4856
product_crowdfunding     4856
round_A                  4856
round_B                  4856
round_C   

In [48]:
raw_data[raw_data['country_code'] == 'ITA'].isnull().sum()

permalink                 0
name                      0
homepage_url              7
category_list            34
market                   34
funding_total_usd         0
status                    3
country_code              0
state_code              308
region                    0
city                      5
funding_rounds            0
founded_at               82
founded_month            82
founded_quarter          82
founded_year             82
first_funding_at          0
last_funding_at           0
seed                      0
venture                   0
equity_crowdfunding       0
undisclosed               0
convertible_note          0
debt_financing            0
angel                     0
grant                     0
private_equity            0
post_ipo_equity           0
post_ipo_debt             0
secondary_market          0
product_crowdfunding      0
round_A                   0
round_B                   0
round_C                   0
round_D                   0
round_E             

##### Strip columns' names

In [40]:
raw_data.columns = raw_data.columns.str.strip()

In [41]:
raw_data.columns

Index(['permalink', 'name', 'homepage_url', 'category_list', 'market',
       'funding_total_usd', 'status', 'country_code', 'state_code', 'region',
       'city', 'funding_rounds', 'founded_at', 'founded_month',
       'founded_quarter', 'founded_year', 'first_funding_at',
       'last_funding_at', 'seed', 'venture', 'equity_crowdfunding',
       'undisclosed', 'convertible_note', 'debt_financing', 'angel', 'grant',
       'private_equity', 'post_ipo_equity', 'post_ipo_debt',
       'secondary_market', 'product_crowdfunding', 'round_A', 'round_B',
       'round_C', 'round_D', 'round_E', 'round_F', 'round_G', 'round_H'],
      dtype='object')